### PR TITLE
Create Linux diagnostic task

### PR DESCRIPTION
**Pull request by jlongever**
@RackHD/corecommitters @uppalk1 @tannoa2 @zyoung51 @keedya 

Dependency on PR https://github.com/RackHD/on-http/pull/345

### CODE REVIEW

File: lib/task-data/tasks/run-emc-diag.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+module.exports = {
+    friendlyName: 'Run EMC Diag',
+    injectableName: 'Task.Os.Run.Emc.Diag',
+    implementsTask: 'Task.Base.Os.Install',
+    options: {
+        kernelFile: 'vmlinuz-4.1.15-diag',
+        initrdFile: 'ramfs.lzma',
+        kernelUri: '{{ api.server }}/common/{{ options.kernelFile }}',
+        initrdUri: '{{ api.server }}/common/{{ options.initrdFile }}',
```
Comment (anhou) [inline lib/task-data/tasks/run-emc-diag.js]: looks a little duplication here. Maybe remove 'kernelUri' and 'initrdUri' then add a field like 'directory: common' or 'owner: common' or some other more reasonable name, and more, add 'uri: {{api.server}}' ' is more  flexible?

Comment (jlongever) [inline lib/task-data/tasks/run-emc-diag.js]: @anhou  This follows the same pattern as `bootstrap-ubuntu.js`.  https://github.com/RackHD/on-tasks/blob/master/lib/task-data/tasks/bootstrap-ubuntu.js

Should we change all of these task definitions?  Is this a gating issue for you?

Comment (anhou) [inline lib/task-data/tasks/run-emc-diag.js]: if you think this is fine, it's just nice to have. but not a gating from me.

Comment (jlongever) [inline lib/task-data/tasks/run-emc-diag.js]: Thanks.

**Pull request merged**